sdk/security/keyvault/azsecrets/autorest.md

```diff
@@ -84,6 +84,12 @@ directive:
     where: $
     transform: return $.replace(/DeletionRecoveryLevel/g, "string");
 
+  # Remove MaxResults parameter
```

Comment:
Is it really so confusing we need to remove it? Could we instead document whatever's confusing about it?

Reply:
I'm down to keep it in, but I think there was something with the service not following the expected behavior? My bad, I can't remember the exact conversation, but I know @heaths was passionate about removing it.

Reply:
Pinged @heaths (he said it was ok if I copied and pasted his response):

"It doesn't do what customers think it does and causes more questions than helps. It's an upper max - max 25 anyway - but each page can contain from 0..MaxResults. Even at 25, that's never too large of a page to handle. It's not like it's 100s or 1000s of results customers may want to chunk. It's utterly useless."

Reply:
So it does what the documentation says it does, but that's seldom helpful? I'd keep it because we prefer accurately representing the service API to rewriting that API on the client side

Reply:
We have already renamed a bunch of stuff because we collectively felt - including our KV service partners - the names are better than what the service model names were. Given MaxResults garnered more questions than known usage, we elided it because it's confusing more than it's helpful, especially when the max value - 25 - is so low it doesn't even help with such small payloads. It's not like we're scoping 1000s of pages to just a 100 at a time to reduce bandwidth. It's 0...25 max items that are each less than 1kb easily.

Reply:
Settled this in offline discussion. We will remove `MaxResults` from Key Vault.